### ISSUE TITLE
 the quality \ the filesize of different libaom versions are quite different.

### ISSUE THREAD

**Issue by Hpsyche** [strukturag/libheif#895]
# Background
Since libaom has optimized the time consumption and image size in version 3.0.0+, we consider upgrading libaom to version 3.6.0;

# Question
Using the same image, under the same heif-enc\lieheif version, it is found that after upgrading libaom the quality and image size are quite different.
## Environment
ubuntu 20.04
libheif version: 1.14.2
## Original image
![ori_gh](https://github.com/strukturag/libheif/assets/37608082/13cff7f9-3e02-46a0-909f-fbec2dfa55e3)
image_size = 17688
## Command
`heif-enc ori.jpg -p quality=50 -A -o dest.avif`
## Result
[dest_avif.zip](https://github.com/strukturag/libheif/files/11733656/dest_avif.zip)
#### aom 2.0.0 version
```
heif-enc -A --list-encoders
aom = AOMedia Project AV1 Encoder v2.0.0
```
image_size = 14514

#### aom 3.6.0 version
```
heif-enc -A --list-encoders
aom = AOMedia Project AV1 Encoder v3.6.0
```
image_size = 7251
The image size and quality of the new version have deteriorated a lot. 
## Confuse
I try to use `heif-enc -A -P`,found that the default parameters are the same. 
So I'm confused that is this normal, or my parameters are wrong ?

**Comment by wantehchang** [comment]
@Hpsyche Thank you for the bug report. I can see the quality difference between aom_2_0_0.avif and aom_3_6_0.avif. I will try to reproduce the results and take a look.

**Comment by wantehchang** [comment]
@Hpsyche @farindk

I found out the reason. This is caused by a bug in libaom v2.0.0 (fixed by the change https://aomedia-review.googlesource.com/c/aom/+/113064). When the bug is present, libaom will use a lower quantizer index (which implies higher quality and larger file size) for the first frame.

I added some debugging printf statements to libaom and libheif. Here is the output for the command `heif-enc ori.jpg -p quality=50 -A -o dest.avif`.

Before the bug fix (at libaom commit 0481a781e262b90f659664135a9ba62bfb59a011):

```
Setting cfg.rc_min_quantizer = 0, cfg.rc_max_quantizer = 63
Calling aom_codec_control(AOME_SET_CPUUSED, 6)
Calling aom_codec_control(AOME_SET_CQ_LEVEL, 32)
active_best_quality = 62, active_worst_quality = 128
q = 62
```

After the bug fix (at libaom commit 8db5ca98460ae4843e657ef53abd26632d6ce07f):

```
Setting cfg.rc_min_quantizer = 0, cfg.rc_max_quantizer = 63
Calling aom_codec_control(AOME_SET_CPUUSED, 6)
Calling aom_codec_control(AOME_SET_CQ_LEVEL, 32)
active_best_quality = 128, active_worst_quality = 128
q = 128
```

Note that the value of `q` is 62 (before) vs 128 (after).

NOTE: The range of `rc_min_quantizer`, `rc_max_quantizer`, and `AOME_SET_CQ_LEVEL` (libaom's external quantizer index) is 0..63. The range of `active_best_quality`, `active_worst_quality`, and `q` (libaom's internal quantizer index) is 0..255.

So, libaom v2.0.0 (which has this bug) with `quality=50` is roughly equivalent to later versions of libaom with `quality=75`.

Here is the output of the command `heif-enc ori.jpg -p quality=75 -A -o dest.avif` with the libaom bug fix:

```
Setting cfg.rc_min_quantizer = 0, cfg.rc_max_quantizer = 63
Calling aom_codec_control(AOME_SET_CPUUSED, 6)
Calling aom_codec_control(AOME_SET_CQ_LEVEL, 16)
active_best_quality = 64, active_worst_quality = 64
q = 64
```

In conclusion, if you have chosen a `quality` value based on experiments with libaom v2.0.0, you will need to use a larger `quality` value with later versions of libaom.

The latest libaom release is v3.6.1.

**Comment by Hpsyche** [comment]
@farindk @wantehchang 
Thank you for your reply. 
That makes sense.It seems that I need to reevaluate the libaom quality standards.

**Comment by farindk** [comment]
@wantehchang Thank you for that thorough analysis.

Probably, it would make sense that libheif compensates for the bug by setting a different `AOME_SET_CQ_LEVEL` when we detect a libaom version <= v2.0.0 

Is there a simple formula for this? You say that old quality=50 is roughly equivalent to new quality=75. I.e. old `CQ_LEVEL=32` is equivalent to new `CQ_LEVEL=16`. Thus, can we simply multiply the `CQ_LEVEL` by 2 for libaom <=v2.0.0 to compensate the bug?

What is the exact libaom version that we would have to check for?

**Comment by wantehchang** [comment]
Dirk: I looked into this. It is not that easy to do what you suggested. Fortunately I found a simple workaround: we can set `cfg.rc_min_quantizer` to `cq_level` for the affected libaom releases to prevent the quantizer index from being lower than `cq_level`.

I also determined that the last libaom release with this bug is v2.0.2.

Please review my pull request https://github.com/strukturag/libheif/pull/901. Thanks.
